structuralcodes/sections/_generic.py

```diff
@@ -0,0 +1,461 @@
+        geometries: t.List[SurfaceGeometry] | MultiPolygon,
+        materials: t.Optional[t.List[Material] | Material] = None
+    ) -> None:
+        '''Creates a compound geometry
+
+        Args:
+            geometries: a list of SurfaceGeometry objects or a shapely
+                        MultiPolygon object (in this case also a list of
+                        materials should be given)
+            materials (optional, default = None): a material (applied
+                        to all polygons) or a list of materials. In this
+                        case the number of polygons should match the number
+                        of materials
+        '''
+
+        if isinstance(geometries, MultiPolygon):
+            # a MultiPolygon is provided
+            if isinstance(materials, Material):
+                # one material is provided. Applied to all polygons
+                pass
+            elif isinstance(materials, list):
+                # the list of materials is provided, one for each polygon
+                pass
+            raise NotImplementedError
+            # x = np.asarray(x, dtype=np.float64)
+            # y = np.asarray(y, dtype=np.float64)
+            # if x.shape != y.shape:
+            #     raise ValueError("x and y must be array-like with the same shape")
+        elif isinstance(geometries, list):
+            # a list of SurfaceGeometry is provided
+            checked_geometries = []
+            for geo in geometries:
+                if isinstance(geo, SurfaceGeometry):
+                    checked_geometries.append(geo)
+                elif isinstance(geo, CompoundGeometry):
+                    for g in geo.geometries:
+                        checked_geometries.append(g)
+            self.geometries = checked_geometries
+
+    # we can add here static methods like
+    # from_dxf
+    # from_ascii
+    # ...
+
+    @property
+    def area(self) -> float:
+        """Return the area of the compund geometry"""
+        area = 0
+        for geo in self.geometries:
+            area += geo.area
+        return area
+
+    # Add split method that call the split for each geometry
+
+    # Add methods for translation and rotation of compound and single geoms
+
+   
+def add_reinforcement(
+    geo: SurfaceGeometry | CompoundGeometry,
+    coords: t.Tuple(float, float),
+    diameter: float,
+    material: Material
+) -> CompoundGeometry:
+    """Adds a reinforcement bar to the geometry
+
+    Proposals:
+    1. Or we can rethink the structure and add:
+        i. Geometry class (group_label -> to be use for filtering)
+        ii. PointGeometry(Geometry) that is a point with a mat
+        iii. SurfaceGeometry(Geometry) that is a polygon with a mat
+        iv. CompoundGeometry(Geometry) that is a set of geometries
+
+    For compound we could implement simple + method so that geo1 + geo2
+    returns a compunt.
+    So this method could become simply:
+
+    bar = Polygon(xxxx)
+    return geo + bar
```

Comment:
Implementing the [`__add__`](https://docs.python.org/3/reference/datamodel.html#emulating-numeric-types) method so that we can add reinforcement by simply `geom + bar` would be a nice feature indeed 👍